mock/channels/channelService.js

```diff
@@ -1,40 +1,138 @@
-class AppChannelService {
-    async joinChannel() {
-        return await window.fdc3.getOrCreateChannel("test-channel");
-    }
+class Fdc3CommandExecutor {
+  //execute commands in order
+  async executeCommands(orderedCommands, config, testId) {
+    let channel;
 
-    async broadcast(contextType, historyItem, channel) {
-        await channel.broadcast({
-            "type": contextType,
-            "name": `History-item-${historyItem}`
-        });
+    for (const command of orderedCommands) {
+      switch (command) {
+        case commands.joinSystemChannelOne: {
+          channel = await this.joinSystemChannelOne();
+          break;
+        }
+        case commands.retrieveTestAppChannel: {
+          channel = await this.retrieveTestAppChannel();
+          break;
+        }
+        case commands.broadcastInstrumentContext: {
+          await this.broadcastContextItem(
+            "fdc3.instrument",
+            channel,
+            config.historyItems, 
+            testId
```

Comment:
`testId` param should be `config.testId`

Reply:
as it was taken as an argument its correct here - as long as it was picked out of context.config earlier to pass in. Regardless, applied the earlier comment so will go this way.